src/commands/codepush/lib/react-native-utils.ts

```diff
@@ -246,6 +246,93 @@ export function runReactNativeBundleCommand(bundleName: string, development: boo
   });
 }
 
+export function runHermesEmitBinaryCommand(bundleName: string, outputFolder: string, sourcemapOutput: string, extraHermesOptions: string[]): Promise<void> {
+  const hermesArgs: string[] = [];
+  const envNodeArgs: string = process.env.CODE_PUSH_NODE_ARGS;
+
+  if (typeof envNodeArgs !== "undefined") {
+      Array.prototype.push.apply(hermesArgs, envNodeArgs.trim().split(/\s+/));
+  }
+
+  Array.prototype.push.apply(hermesArgs, [
+      "-emit-binary",
+      "-out", path.join(outputFolder, bundleName + ".hbc"),
+      path.join(outputFolder, bundleName),
+      ...extraHermesOptions,
+  ]);
+
+  if (sourcemapOutput) {
+      hermesArgs.push("-output-source-map");
+  }
+
+  out.text(chalk.cyan("Converting JS bundle to byte code via Hermes, running command:\n"));
+  const hermesProcess = spawn(path.join("node_modules", "hermesvm", getHermesOSBin(), "hermes"), hermesArgs);
+  out.text(`${path.join("node_modules", "hermesvm", getHermesOSBin(), "hermes")} ${hermesArgs.join(" ")}`);
+
+  return new Promise<void>((resolve, reject) => {
+      hermesProcess.stdout.on("data", (data: Buffer) => {
+        out.text(data.toString().trim());
+      });
+
+      hermesProcess.stderr.on("data", (data: Buffer) => {
+        // Since hermes is printing lots of messages on stderr, we skip anything here.
```

Comment:
After a little study, those message I've mentioned is actually warnings on Javascript, which is now printed to `stderr` once `--debug` is there, as for other errors, it will now print to `console.error`.